Formula/gtk+4.rb

```diff
@@ -0,0 +1,149 @@
+class Gtkx4 < Formula
+  desc "Toolkit for creating graphical user interfaces"
+  homepage "https://gtk.org/"
+  url "https://download.gnome.org/sources/gtk/4.0/gtk-4.0.0.tar.xz"
+  sha256 "d46cf5b127ea27dd9e5d2ff6ed500cb4067eeb2cb1cd2c313ccde8013b0b9bf9"
+  license "LGPL-2.0-or-later"
+
+  livecheck do
+    url :stable
+    regex(/gtk\+[._-](4\.([0-8]\d*?)?[02468](?:\.\d+)*?)\.t/i)
+  end
+
+  depends_on "docbook" => :build
+  depends_on "docbook-xsl" => :build
+  depends_on "gobject-introspection" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "gdk-pixbuf"
+  depends_on "glib"
+  depends_on "graphene"
+  depends_on "hicolor-icon-theme"
+  depends_on "libepoxy"
+  depends_on "pango"
+  depends_on "sassc"
+
+  uses_from_macos "libxslt" => :build # for xsltproc
+
+  # submitted upstream at https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/2962
+  patch :DATA
+
+  def install
+    args = std_meson_args + %w[
+      -Dx11-backend=false
+      -Dmacos-backend=true
+      -Dgtk_doc=false
+      -Dman-pages=true
+      -Dintrospection=enabled
+      -Dbuild-examples=false
+      -Dbuild-tests=false
+    ]
+
+    # ensure that we don't run the meson post install script
+    ENV["DESTDIR"] = "/"
+
+    # Find our docbook catalog
+    ENV["XML_CATALOG_FILES"] = "#{etc}/xml/catalog"
+
+    mkdir "build" do
+      system "meson", *args, ".."
+      system "ninja", "-v"
+      system "ninja", "install", "-v"
+    end
+  end
+
+  def post_install
+    system "#{Formula["glib"].opt_bin}/glib-compile-schemas", "#{HOMEBREW_PREFIX}/share/glib-2.0/schemas"
+    system bin/"gtk4-update-icon-cache", "-f", "-t", "#{HOMEBREW_PREFIX}/share/icons/hicolor"
+    system "#{Formula["glib"].opt_bin}/gio-querymodules #{HOMEBREW_PREFIX}/lib/gtk-4.0/4.0.0/immodules.cache"
+  end
+
+  test do
+    (testpath/"test.c").write <<~EOS
+      #include <gtk/gtk.h>
+
+      int main(int argc, char *argv[]) {
+        gtk_disable_setlocale();
+        return 0;
+      }
+    EOS
+    cairo = Formula["cairo"]
+    fontconfig = Formula["fontconfig"]
+    freetype = Formula["freetype"]
+    fribidi = Formula["fribidi"]
+    gdk_pixbuf = Formula["gdk-pixbuf"]
+    gettext = Formula["gettext"]
+    glib = Formula["glib"]
+    graphene = Formula["graphene"]
+    graphite2 = Formula["graphite2"]
+    libffi = Formula["libffi"]
+    harfbuzz = Formula["harfbuzz"]
+    libepoxy = Formula["libepoxy"]
+    libpng = Formula["libpng"]
+    pango = Formula["pango"]
+    pcre = Formula["pcre"]
+    pixman = Formula["pixman"]
+    flags = %W[
```

Comment:
Can't you use `pkg-config` to generate at least some of these flags more compactly?

Reply:
Yes, and in fact this is derived from its output but there was a time when using `pkg-config` in test blocks was very much frowned upon, so this approach was necessary.

Reply:
I can't remember that time so please use pkg-config 😁